cmd/strace.go

```diff
@@ -0,0 +1,119 @@
+package cmd
+
+import (
+	"os/exec"
+	"runtime"
+	"syscall"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+var straceCommand = &cli.Command{
+	Name:   "strace",
+	Usage:  "Run strace of a program e.g. strace ps aux",
+	Action: strace,
+}
+
+func strace(c *cli.Context) error {
+
+	// Pins the goroutine to the thread
+	// which is necessary for ptrace to work correctly
+	// (probably not needed) in non-concurrent context though,
```

Comment:
Remove `)` after `needed`.  
Why doesn't ptrace work correctly if we don't do this?

Reply:
I cannot explain it fully, they recommend it here: https://golang.org/pkg/syscall/#SysProcAttr 

My guess is that in highly concurrent context the goroutine can be scheduled using different threads. If that happens, then you might have issues with respect to which thread is attached where through ptrace. As per http://man7.org/linux/man-pages/man2/ptrace.2.html: 

      A tracee first needs to be attached to the tracer.  Attachment and
       subsequent commands are per thread: in a multithreaded process, every
       thread can be individually attached to a (potentially different)
       tracer, or left not attached and thus not debugged.  Therefore,
       "tracee" always means "(one) thread", never "a (possibly
       multithreaded) process".  Ptrace commands are always sent to a
       specific tracee using a call of the form